Tests/Common/Util/StringAnyEncodable.swift

```diff
@@ -96,10 +96,38 @@ class StringAnyEncodableTest: UnitTest {
             "boolean": true,
             "array": [1, 2, 4],
             "dict": ["test": "value"] as [String: Any],
-            "hashables": [AnyHashable("color"): [AnyHashable("Red"), AnyHashable("Green"), AnyHashable("Blue")]] as [AnyHashable: [AnyHashable]]
+            "dictWithArray": ["color": ["Red", "Green", "Blue"]] as [String: [Any]]
         ] as [String: Any]
 
-        let json = StringAnyEncodable(data)
+        let json = StringAnyEncodable(logger: log, data)
+
+        guard let actual = jsonAdapter.toJson(json) else {
+            XCTFail("couldn't encode to JSON")
+            return
+        }
+
+        guard let result: DummyData = jsonAdapter.fromJson(actual) else {
+            XCTFail("data did not decoded to a DummyData object")
+            return
+        }
+
+        XCTAssertEqual(expect, result)
+    }
+
+    func test_stringanyencodable_encodes_complex_datafromwrappers() {
```

Comment:
I am not sure that this test function adds any value to our test suite?

The idea that a value gets converted to `AnyHashable` should not break our SDK's code so I can't think of a reason to test against it. 

A value has to be Hashable in order to be stored in an array. `AnyHashable` is identical to `Hashable`, but generic. 

Therefore, since you added `"dictWithArray": ["color": ["Red", "Green", "Blue"]] as [String: [Any]]` to the test function above, I think that tests `AnyHashable` well enough and we can delete this test function.